Restructuring HubDocs documentation

micokoch opened the pull request.

harryhoch commented:
Some suggestions:

1."Who we are": Under "funding support" can we add MIDAS Coordination Center NIGMS #U24GM132013 for Lucie's effort?
2. Update the copyright to 2023.
3. "Setting up a simple modeling hub" - the conversation of anaconda & R might be a bit confusing for those who don't know anaconda.
4.  The documentation regularly mentions "origin_date" without really defining it.  A clear example of the relationships between the different dates might help.
5.  "Configuring tasks" -  the output type descriptions might benefit from a bit more detail. I am not sure what the "NA" value for optional means.
6.  "Uploading and validating config files" - I couldn't get the code to work at first, as I didn't realize that the routine would append "hub-config" to the path. Clearer directions might be needed.
7.  "Using the modeling hubs" - the vignette link goes to the source for the vignette. It should probably link to a PDF/html  version.
8.  It might make sense to introduce hub configuration files before modeling tasks
9.  in " Defining modeling tasks", some of the task definition language is confusing. At one point, it says that the task id variables for example 1 are target, location, origin_date and horizon.  It then goes on to use example 3 as demonstrating how multiple variables can solve as target keys together. Isn't that exactly what's going on in example 1?
10. This page is also inconsistent with the example, in terms of whether or not "horizon" is a task id
11.  It would be good to have a clearer description of the difference between the model metadata and the model abstracts.
12.  "Model output"  - an example of a joint output might help. 
13.  Under software, add hubviz.

Writing documentation is very difficult - this is shaping up very nicely...

micokoch commented:
Thanks for the suggestions, Harry! I will work on them and let you know once these issues are addressed.

harryhoch commented:
you're welcome!

nickreich reviewed (changes requested):
The quickstarts look great! I have a few mostly minor comments, enumerated here.

**Quickstart-admin**
- [ ] on the intro page, could we provide a link to the "key definitions" page? a few pages in we refer to "rounds" but if someone starts at this intro page, they might not know how to find out what "rounds" are.
- [ ] on setting-up page, the top item is "configuring your modeling hub" but it really seems to mostly be about getting set up with the right software. maybe that should have its own header? or maybe even the software part should be moved to the prior page on "getting started"? 
- [ ] on the "configuring your modeling hub", there is a link to the more detailed information that is available on the "Hub configuration files" page at the very top, but I'd suggest also putting it lower down where the admin.json file is being looked at, as the info at that link provides more information about all possible fields and what each field is. 
- [ ] on configuring tasks step 5.1 it says "...that no origin dates are required..." maybe add that no specific origin_date is required because every submission will have a different origin_date. E.g. some locations will be required because they must be present in every submission, but origin date is different since it changes with each submission.
- [ ] Step 5.5 provide a link to the more detailed definition of what an output type is: https://hubdocs.readthedocs.io/en/latest/format/model-output.html#formats-of-model-output
- [ ] Step 5.6: the text  "target_keys"  in the phrase ""target_keys" must match " is not visible because the link text is pink and the highlight is also pink. 
- [ ] Step 5.7: mention explicitly that the submissions due is used by hubValidations in validating submission files.
- [ ] the updating and validating page has a really nice checklist. Could we also include this checklist as an outline on the introduction page for the admin quickstart?
- [ ] on the using-hub page, maybe provide a link to an example Technical README file on another hub as an example?
- [ ] links to hubvalidations and hubEnsembles packages and vignettes should be to pkgdown site, not the GitHub repo

**Quickstart-using data**

- [ ] This section appears to be empty. Maybe we should just remove it for now from the documentation if we aren't planning on populating it with content right away?

micokoch commented:
To keep track of all these issues I'm making a tasks list:

**Pending:**
- [ ] "Configuring tasks" - the output type descriptions might benefit from a bit more detail. I am not sure what the "NA" value for optional means.
- [ ] "Uploading and validating config files" - I couldn't get the code to work at first, as I didn't realize that the routine would append "hub-config" to the path. Clearer directions might be needed.
- [ ] "Using the modeling hubs" - the vignette link goes to the source for the vignette. It should probably link to a PDF/html version. _Note: I will ask Lucie about this._
- [ ] It might make sense to introduce hub configuration files before modeling tasks.
- [ ] in " Defining modeling tasks", some of the task definition language is confusing. At one point, it says that the task id variables for example 1 are target, location, origin_date and horizon. It then goes on to use example 3 as demonstrating how multiple variables can solve as target keys together. Isn't that exactly what's going on in example 1?
- [ ] This page is also inconsistent with the example, in terms of whether or not "horizon" is a task id.
- [ ] It would be good to have a clearer description of the difference between the model metadata and the model abstracts.
- [ ] "Model output" - an example of a joint output might help.
- [ ] on setting-up page, the top item is "configuring your modeling hub" but it really seems to mostly be about getting set up with the right software. maybe that should have its own header? or maybe even the software part should be moved to the prior page on "getting started"?
- [ ] on the "configuring your modeling hub", there is a link to the more detailed information that is available on the "Hub configuration files" page at the very top, but I'd suggest also putting it lower down where the admin.json file is being looked at, as the info at that link provides more information about all possible fields and what each field is.
- [ ] Step 5.5 provide a link to the more detailed definition of what an output type is: https://hubdocs.readthedocs.io/en/latest/format/model-output.html#formats-of-model-output
- [ ] the updating and validating page has a really nice checklist. Could we also include this checklist as an outline on the introduction page for the admin quickstart?
- [ ] on the using-hub page, maybe provide a link to an example Technical README file on another hub as an example?

**Completed:**
- [x] "Who we are": Under "funding support" can we add MIDAS Coordination Center NIGMS #U24GM132013 for Lucie's effort? _Note: @harryhoch - I asked Lucie, and put the start of funding in June 2022_
- [x] Update the copyright to 2023.
- [x] "Setting up a simple modeling hub" - the conversation of anaconda & R might be a bit confusing for those who don't know anaconda. _Note: agree and deleted that part._
- [x] The documentation regularly mentions "origin_date" without really defining it. A clear example of the relationships between the different dates might help. _Note: explanations were added [here](https://hubdocs--63.org.readthedocs.build/en/63/quickstart-hub-admin/tasks-config.html#step-5-examine-the-new-tasks-json-file)._
- [x] Under software, add hubviz.
- [x] on the intro page, could we provide a link to the "key definitions" page? a few pages in we refer to "rounds" but if someone starts at this intro page, they might not know how to find out what "rounds" are.
- [x] on configuring tasks step 5.1 it says "...that no origin dates are required..." maybe add that no specific origin_date is required because every submission will have a different origin_date. E.g. some locations will be required because they must be present in every submission, but origin date is different since it changes with each submission.
- [x] Step 5.6: the text "target_keys" in the phrase ""target_keys" must match " is not visible because the link text is pink and the highlight is also pink. _Note: I changed the inline code font to dark purple (using custom css)._
- [x] Step 5.7: mention explicitly that the submissions due is used by hubValidations in validating submission files.
- [x] links to hubvalidations and hubEnsembles packages and vignettes should be to pkgdown site, not the GitHub repo
- [x] This section appears to be empty. Maybe we should just remove it for now from the documentation if we aren't planning on populating it with content right away?

nickreich commented:
There are some blank pages:

- [ ] getting help
- [ ] release notes
- [ ] future ideas and suggestions
- [ ] others too... please click through all links on the PR draft site to see them.

mzorn-58 commented:
@nickreich @elray1 @LucieContamin @lshandross @micokoch @harryhoch @annakrystalli 

I am working on the following change suggested:
 _

> _**_in " [Defining modeling tasks"](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/tasks.html), some of the task definition language is confusing. At one point, it says that the task id variables for example 1 are target, location, origin_date and horizon. It then goes on to use example 3 as demonstrating how multiple variables can solve as target keys together. Isn't that exactly what's going on in example 1?_**_

_
 
 Here is the verbiage referred to:

> We note that some task ID variables are special in that they conceptually define a modeling “target” (these are referred to in the [tasks metadata](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/hub-config.html#tasks-metadata) as a target_key). In our [Running Example 1](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/intro-data-formats.html#running-examples), the task ID variables are target, location, origin_date, and horizon. In this example, target is the target key. In other examples, (such as [Running Example 3](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/intro-data-formats.html#running-examples)) more than one variable can serve as target keys together.
 

-  For example 1, horizon is not a task ID variable.  Confirming that we can remove that from the text.
-  I think there is a confusion between task id variables and target key.  Does the text clarify it for you?

> We note that some task ID variables are special in that they conceptually define a modeling “target” (these are referred to in the [tasks metadata](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/hub-config.html#tasks-metadata) as a target_key). In our [Running Example 1](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/intro-data-formats.html#running-examples), the task ID variables are target, location, origin_date. In this e
xample, `target` is the **target key** and can only take on one value "inc covid hosp". In other examples, (such as [Running Example 3](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/intro-data-formats.html#running-examples)) more than one variable can serve as target keys together. In example 3, both `outcome_variable` and `outcome_measure` make up the target keys.

LucieContamin commented:
> @nickreich @elray1 @LucieContamin @lshandross @micokoch @harryhoch @annakrystalli
> 
> I am working on the following change suggested: _
> 
> > _**_in " [Defining modeling tasks"](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/tasks.html), some of the task definition language is confusing. At one point, it says that the task id variables for example 1 are target, location, origin_date and horizon. It then goes on to use example 3 as demonstrating how multiple variables can solve as target keys together. Isn't that exactly what's going on in example 1?_**_
> 
> _
> 
> Here is the verbiage referred to:
> 
> > We note that some task ID variables are special in that they conceptually define a modeling “target” (these are referred to in the [tasks metadata](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/hub-config.html#tasks-metadata) as a target_key). In our [Running Example 1](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/intro-data-formats.html#running-examples), the task ID variables are target, location, origin_date, and horizon. In this example, target is the target key. In other examples, (such as [Running Example 3](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/intro-data-formats.html#running-examples)) more than one variable can serve as target keys together.
> 
>     * For example 1, horizon is not a task ID variable.  Confirming that we can remove that from the text.
> 
>     * I think there is a confusion between task id variables and target key.  Does the text below clarify it for you?
> 
> 
> > We note that some task ID variables are special in that they conceptually define a modeling “target” (these are referred to in the [tasks metadata](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/hub-config.html#tasks-metadata) as a target_key). In our [Running Example 1](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/intro-data-formats.html#running-examples), the task ID variables are target, location, origin_date. In this e
> > xample, `target` is the **target key** and can only take on one value "inc covid hosp". In other examples, (such as [Running Example 3](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/intro-data-formats.html#running-examples)) more than one variable can serve as target keys together. In example 3, both `outcome_variable` and `outcome_measure` make up the target keys.

I like the new text. 
I also notice that the Running Example 3 link to the US COVID-19 Scenario Modeling Hub, but in the hub we use only one column `target`, we never split it so that example title need to be updated too. Happy to help with that if necessary

mzorn-58 commented:
@LucieContamin Thanks! How is this for a new title for Example 3?

> Example 3: Multiple target keys

mzorn-58 commented:
@LucieContamin @nickreich @harryhoch @elray1 @lshandross @annakrystalli 
Next question, regarding task item

>  It would be good to have a clearer description of the difference between the model metadata and the model abstracts.

How about adding the  text below under **purpose** on both the[ `model-metadata`](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/model-metadata.html) and [`model-abstract`](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/model-abstracts.html) pages?

> Note: [Model metadata](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/model-metadata.html) is a required feature of a hub and describes the characteristics of models contributing to a Hub, while [Model abstracts](https://hubdocs--63.org.readthedocs.build/en/63/user-guide/model-abstracts.html#) is an optional feature of a hub that is useful for keeping track of round-specific detailed narrative descriptions modeling methods and results.

LucieContamin commented:
> @LucieContamin task item
> 
> > "Using the modeling hubs" - the vignette link goes to the source for the vignette. It should probably link to a PDF/html version. Note: Lucie said she will update it as soon as the hubVis is running.
> 
> Just confirming that vingette is not ready yet.

Yes it is not ready yet, but as soon as it's updated, I will update the link to an HTML or PDF version.
